api/src/messaging/receive.py

```diff
@@ -0,0 +1,77 @@
+from channels.db import database_sync_to_async
+from .models import Message
+
+
+async def receive_message(consumer, text_data_json: dict) -> dict:
```

Comment:
I think this function should either:
1. Take in 1 parameter that is a dictionary with ONLY the data this function needs (which means not passing the whole instance of the ChatConsumer)
2. Take in multiple parameters, each parameter for the data it needs (so text_data_json["message_body"] would be 1 parameter, and so on)

I think option 1 makes the most sense.